backend/app/db/crud/document.py

```diff
@@ -600,6 +549,194 @@ def write_metadata(
+            name=d.name,
+            description=d.description,
+            country_code=g.value,
+            country_name=g.display_value,
+            publication_ts=d.publication_ts,
+        )
+        for _, d, g in (
+            db.query(Association, Document, Geography)
+            .filter(Association.document_id_from == document_id)
+            .join(Document, Association.document_id_to == Document.id)
+            .join(Geography, Document.geography_id == Geography.id)
+        ).all()
+    )
+    # if the above query has results, it contains a "master doc" for a group,
+    # so collect all children
+    related_to_master_documents = set()
+    # for alpha this list should have at most one entry
+    for related_doc in related_documents_to:
+        related_to_master_documents |= set(
+            DocumentOverviewResponse(
+                document_id=d.id,
+                name=d.name,
+                description=d.description,
+                country_code=g.value,
+                country_name=g.display_value,
+                publication_ts=d.publication_ts,
+            )
+            for _, d, g in (
+                db.query(Association, Document, Geography)
+                .filter(Association.document_id_to == related_doc.document_id)
+                .filter(Association.document_id_from != document_id)
+                .join(Document, Association.document_id_from == Document.id)
+                .join(Geography, Document.geography_id == Geography.id)
+            ).all()
+        )
+    # finally find all (child) documents that refer to this document_id
+    related_documents_from = set(
+        DocumentOverviewResponse(
+            document_id=d.id,
+            name=d.name,
+            description=d.description,
+            country_code=g.value,
+            country_name=g.display_value,
+            publication_ts=d.publication_ts,
+        )
+        for _, d, g in (
+            db.query(Association, Document, Geography)
+            .filter(Association.document_id_to == document_id)
+            .join(Document, Association.document_id_from == Document.id)
+            .join(Geography, Document.geography_id == Geography.id)
+        ).all()
+    )
+    return related_documents_to | related_to_master_documents | related_documents_from
+
+
+def _get_related_documents(db: Session, document_id: int):
+    """Gets all the other documents this document is related to.
+
+    TODO: return this as structured, as at the moment we return a flat list
+    """
+    print(f"### get_related_documents for: {document_id}")
+    query_all_relationships_of_document = db.query(
+        DocumentRelationship.relationship_id
+    ).filter(DocumentRelationship.document_id == document_id)
+
+    print(
+        f"### get_related_documents searching: {db.query(DocumentRelationship).count()}"
+    )
+    print(
+        f"### get_related_documents found relns: {query_all_relationships_of_document.count()}"
+    )
```

Comment:
I think we need to be getting rid of the debug print statements...

Reply:
oh :hankey:

Reply:
I commented out the entire function - as this is for the next phase